spec/function.dd

```diff
@@ -1542,6 +1542,82 @@ class C
         $(P Template functions, auto functions, nested functions and lambdas can deduce
         the `return scope` attribute.)
 
+$(H3 $(LNAME2 ref-return-scope-parameters, Ref Return Scope Parameters))
+
+        $(P Parameters marked as `ref return scope` come in two forms:)
+
+---
+U xerxes(ref return scope V v);      // (1) ref and return scope
+ref U xerxes(ref return scope V v);  // (2) return ref and scope
+---
+
+        $(P The first form attaches the `return` to the `scope`, and has
+        $(LINK2 #return-scope-parameters, return scope parameter) semantics
+        for the value of the `ref` parameter.)
+
+        $(P The second form attaches the `return` to the `ref`, and has
+        $(LINK2 #return-ref-parameters, return ref parameter) semantics
+        with additional
+        $(LINK2 https://dlang.org/spec/memory-safe-d.html#scope-return-params, scope parameter)
+        semantics.)
+
+        $(P Although a struct constructor returns a reference to the instance
+        being constructed, it is treated as form (1).)
+
+        $(P The lexical order of the attributes `ref`, `return`, and `scope` is not significant.)
+
+        $(P It is not possible to have both `return ref` and `return scope` semantics
+        for the same parameter.)
+
+---
+@safe:
+
+struct S
+{
+    this(return scope   ref int* p) { ptr = p; }
+
+    int  val;
+    int* ptr;
+}
+
+int* foo1(return scope   ref S s);
+int  foo2(return scope   ref S s);
+
+ref int* foo3(return ref   scope S s);
+ref int  foo4(return ref   scope S s);
```

Comment:
Please avoid aligning the code. It will just require more work if new examples are added.

Reply:
I disagree, in this particular case it really helps to get the point across.